### PR TITLE
Telemetry - change NetworkState from string to json

### LINKED ISSUE #2300: "network_state" value in telemetry is string, should be json
`network_state` value in telemetry is string, it should be json, in `"msg":"system.interval"`

Need to check whether this applies to any other values in other messages.

eg: 

```json
"network_state":"{\"peerId\":\"QmaGMX8QCeeiq79YiHc7ndR2npwbk2HdcmFFkYWLBHmznu\",\"listenedAddresses\":[\"/ip4/0.0.0.0/tcp/30333\"],\"externalAddresses\":[\
...
```

### PR DESCRIPTION
**Pull request by mattrutherford**
Closes #2300

### CODE REVIEW

File: core/network-libp2p/src/lib.rs [hunk 1]
```diff
@@ -16,6 +16,9 @@
 
 //! Networking layer of Substrate.
 
+#[macro_use]
+extern crate slog_derive;
```
Comment (bkchr) [inline core/network-libp2p/src/lib.rs]: Why do we need this?

Comment (mattrutherford) [reply core/network-libp2p/src/lib.rs]: Because we want to derive SerdeValue for NetworkState

File: core/network-libp2p/Cargo.toml [hunk 2]
```diff
@@ -29,6 +29,10 @@ tokio-timer = "0.2"
 unsigned-varint = { version = "0.2.1", features = ["codec"] }
 void = "1.0"
 
+slog = { version = "^2", features = ["nested-values"] }
+slog_derive = "0.1.1"
+erased-serde = "0.3.9"
```
Comment (bkchr) [inline core/network-libp2p/Cargo.toml]: Why do we need this? ^^

File: core/network-libp2p/src/lib.rs (hunk 1 above)
Comment (bkchr) [reply core/network-libp2p/src/lib.rs]: Could be drop this line and include the macro where it is required, like it is done for the rest?

File: core/network-libp2p/Cargo.toml (hunk 2 above)
Comment (mattrutherford) [reply core/network-libp2p/Cargo.toml]: fyi: https://docs.rs/slog_derive/0.1.1/slog_derive/?search=#the-serdevalue-derive

Comment (bkchr) [reply core/network-libp2p/Cargo.toml]: Ohh... What for a shitty derive implementation..

File: core/network-libp2p/src/lib.rs (hunk 1 above)
Comment (mattrutherford) [reply core/network-libp2p/src/lib.rs]: sounds fair

**Comment by tomaka** [comment]
Fwiw we should ping @maciejhirsz when changes are made to the telemetry.

**Pull request merged**